src/utilcode/clrhost_nodependencies.cpp

```diff
@@ -383,6 +394,10 @@ FastFreeInProcessHeapFunc __ClrFreeInProcessHeap = (FastFreeInProcessHeapFunc) C
 
 const NoThrow nothrow = { 0 };
 
+#ifdef HAS_ADDRESS_SANITIZER
+// use standard heap functions for address santizier
```

Comment:
It seems that using standard heap functions would change the behavior of the new. Our new throws exception derived from our Exception type and the runtime can handle it at some points. The default new throws a completely different exception.